### ISSUE TITLE
Implement support for installers requiring sudo password

### ISSUE THREAD

**Issue by louchebem06** [buresdv/Cork#164]
# Bug Report

## Mandatory Information
**Describe the bug**
Cork block during the installation of XQuartz, but it works with cli.

**To Reproduce**
Trying to install XQuartz with Cork.

**Screenshots**

![Capture d’écran 2023-09-10 à 18 00 23](https://github.com/buresdv/Cork/assets/34492446/f5f047d9-cd6a-4f17-a4d2-04140856ba49)
![Capture d’écran 2023-09-10 à 18 00 29](https://github.com/buresdv/Cork/assets/34492446/80ed67f4-efe6-4e6c-aa70-3d615047c980)
![Capture d’écran 2023-09-10 à 18 02 49](https://github.com/buresdv/Cork/assets/34492446/de2870c7-597a-4905-9ca6-6cb729c018c9)

**System and Cork Information:**
 - macOS version: 13.5.2
 - Cork Version: 1.2(43)
 - Did you compile Cork yourself?: yes

## Optional Information

**Additional context**
- Output xcode after search xquartz and click install: [log.txt](https://github.com/buresdv/Cork/files/12569765/log.txt)
- I have the impression that sudo rights are not being requested and that they might be needed.

# Checklist
- [x] I filled everything out under **Mandatory Information**
- [x] This bug affects the UI and I have included a screenshot

**Comment by buresdv** [comment]
Man, I have to fix my GitHub notifications, I didn't see this issue. Let me try to see what's up

**Comment by buresdv** [comment]
Can confirm this is because of sudo permissions being required, as per this line:
```
sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
```

This is a long-standing issue that's been around since the beginning, and it's very difficult to fix. Apple provides no documentation on how to elevate permissions for Tasks in Swift, the only example being in Objective-C, which I can't use. The other option is creating a Privileged Helper, which is something way too advanced. 

So, this issue is something I have in mind, but can't fix for now. But I'm doing my best to figure it out

**Comment by louchebem06** [comment]
I don't know if it can help: https://gist.github.com/KyLeggiero/6619bbba8ebb09e6252459521e01ec4b

**Comment by buresdv** [comment]
I implemented permissions escalation, but it turns out that Brew gets pissy when you do that:
```
Error: Running Homebrew as root is extremely dangerous and no longer supported.
As Homebrew does not drop privileges on installation you would be giving all
build scripts full access to your system.
```

So we need a different approach :/

**Comment by buresdv** [comment]
Temporary solution implemented in #184
